account_reconcile_model_multicompany_propagate/models/account_reconcile_model.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2023 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+import logging
+from collections import defaultdict
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountReconcileModel(models.Model):
+    _inherit = "account.reconcile.model"
+
+    def _propagated_fields(self):
+        """Fields to propagate to other companies."""
+        return [
+            "rule_type",
+            "auto_reconcile",
+            "to_check",
+            "matching_order",
+            "match_text_location_label",
+            "match_text_location_note",
+            "match_text_location_reference",
+            "match_nature",
+            "match_amount",
+            "match_amount_min",
+            "match_amount_max",
+            "match_label",
+            "match_label_param",
+            "match_note",
+            "match_note_param",
+            "match_transaction_type",
+            "match_transaction_type_param",
+            "match_same_currency",
+            "allow_payment_tolerance",
+            "payment_tolerance_param",
+            "payment_tolerance_type",
+            "match_partner",
+            "match_partner_ids",
+            "match_partner_category_ids",
+            "past_months_limit",
+            "number_entries",
+        ]
+
+    def propagate_multicompany_account_reconcile_model(self):
```

Comment:
Just to make it a bit shorter and clearer:

```suggestion
    def propagate_to_other_companies(self):
```

IMHO it'd be good to split this method into smaller methods. However it's just a suggestion, it's ok as it is.

Reply:
You are right. Method renamed and splitted.